### PR TITLE
all: rename GCP and AWS SQL packages to use cloud provider prefixes BREAKING_CHANGE_OK

### LINKED ISSUE #2357: mysql/postgres: reconsider package names
E.g., maybe `cloudsql` should be `gcpcloudsql`.

https://github.com/google/go-cloud/blob/master/internal/docs/design.md#package-naming-conventions

### PR DESCRIPTION
**Pull request by zombiezen**
Fixes #2357

### CODE REVIEW

**Comment by zombiezen** [comment]
This will trivially conflict with #2392, but I'll deal with the merge conflicts whichever order they get merged in.

File: internal/website/content/ref/sql.md [hunk 1]
```diff
@@ -25,14 +25,15 @@ https://godoc.org/gocloud.dev/postgres.
 
 For MySQL:
 
-* [GCP Cloud SQL](https://godoc.org/gocloud.dev/mysql/cloudmysql)
-* [AWS RDS](https://godoc.org/gocloud.dev/mysql/rdsmysql)
+* [GCP Cloud SQL](https://godoc.org/gocloud.dev/mysql/gcpmysql)
```
Comment (polinasok) [inline internal/website/content/ref/sql.md]: Not to confuse with the references below, what about 
GCP Cloud SQL for MySQL (as in https://cloud.google.com/sql/docs/mysql/)
or
Google Cloud MySQL (as in https://www.google.com/search?q=google+cloud+mysql)

File: internal/website/content/ref/sql.md [hunk 2]
```diff
@@ -25,14 +25,15 @@ https://godoc.org/gocloud.dev/postgres.
...
+* [AWS RDS](https://godoc.org/gocloud.dev/mysql/awsmysql)
+* [Azure Database](https://godoc.org/gocloud.dev/mysql/azuremysql)
 * [On-Premise](https://godoc.org/gocloud.dev/mysql) (or locally hosted)
 
 For PostgreSQL:
 
-* [GCP Cloud SQL](https://godoc.org/gocloud.dev/postgres/cloudpostgres)
-* [AWS RDS](https://godoc.org/gocloud.dev/postgres/rdspostgres)
+* [GCP Cloud SQL](https://godoc.org/gocloud.dev/postgres/gcppostgres)
```
Comment (polinasok) [inline internal/website/content/ref/sql.md]: same as above

File: internal/website/content/ref/sql.md [hunk 3]
```diff
@@ -25,14 +25,15 @@ https://godoc.org/gocloud.dev/postgres.
...
+* [AWS RDS](https://godoc.org/gocloud.dev/postgres/awspostgres)
```
Comment (polinasok) [inline internal/website/content/ref/sql.md]: same as above

File: internal/website/content/ref/sql.md [hunk 4]
```diff
@@ -25,14 +25,15 @@ https://godoc.org/gocloud.dev/postgres.
 
 For MySQL:
 
-* [GCP Cloud SQL](https://godoc.org/gocloud.dev/mysql/cloudmysql)
-* [AWS RDS](https://godoc.org/gocloud.dev/mysql/rdsmysql)
+* [GCP Cloud SQL](https://godoc.org/gocloud.dev/mysql/gcpmysql)
+* [AWS RDS](https://godoc.org/gocloud.dev/mysql/awsmysql)
```
Comment (polinasok) [inline internal/website/content/ref/sql.md]: Same as above:
AWS RDS for MySQL
AWS RDS MySQL
AWS MySQL
(https://www.google.com/search?q=aws+mysql)

File: samples/go.sum [hunk 5]
```diff
@@ -213,6 +213,7 @@ github.com/kr/pretty v0.1.0/go.mod h1:dAy3ld7l9f0ibDNOQOHHMYYIIbhfbHSm3C4ZsoJORN
 github.com/kr/pty v1.1.1/go.mod h1:pFQYn66WHrOpPYNljwOMqo10TkYh1fy3cYio2l3bCsQ=
 github.com/kr/text v0.1.0 h1:45sCR5RtlFHMR4UwH9sdQ5TC8v0qDQCHnXt+kaKSTVE=
 github.com/kr/text v0.1.0/go.mod h1:4Jbv+DJW3UT/LiOwJeYQe1efqtUx/iVham/4vfdArNI=
+github.com/lib/pq v1.1.1 h1:sJZmqHoEaY7f+NPP8pgLB/WxulyR3fewgCM2qaSlBb4=
```
Comment (vangent) [inline samples/go.sum]: Is this needed?

File: samples/guestbook/inject_aws.go [hunk 6]
```diff
@@ -64,9 +64,9 @@ func awsBucket(ctx context.Context, cp awsclient.ConfigProvider, flags *cliFlags
 
 // openAWSDatabase is a Wire provider function that connects to RDS based on
```
Comment (vangent) [inline samples/guestbook/inject_aws.go]: Maybe update comment?

File: samples/guestbook/inject_gcp.go [hunk 7]
```diff
@@ -66,9 +66,9 @@ func gcpBucket(ctx context.Context, flags *cliFlags, client *gcp.HTTPClient) (*b
 
 // openGCPDatabase is a Wire provider function that connects to Cloud SQL
```
Comment (vangent) [inline samples/guestbook/inject_gcp.go]: Same?

File: internal/website/content/ref/sql.md (hunk 1 above)
Comment (zombiezen) [reply internal/website/content/ref/sql.md]: Done.

File: internal/website/content/ref/sql.md (hunk 4 above)
Comment (zombiezen) [reply internal/website/content/ref/sql.md]: Done.

File: internal/website/content/ref/sql.md (hunk 2 above)
Comment (zombiezen) [reply internal/website/content/ref/sql.md]: Done.

File: internal/website/content/ref/sql.md (hunk 3 above)
Comment (zombiezen) [reply internal/website/content/ref/sql.md]: Done.

File: samples/go.sum (hunk 5 above)
Comment (zombiezen) [reply samples/go.sum]: No. Removed.

File: samples/guestbook/inject_aws.go (hunk 6 above)
Comment (zombiezen) [reply samples/guestbook/inject_aws.go]: Done.

File: samples/guestbook/inject_gcp.go (hunk 7 above)
Comment (zombiezen) [reply samples/guestbook/inject_gcp.go]: Done.

**Pull request merged**